### ISSUE TITLE
Linter gets upset with Regexp in type

### ISSUE THREAD

**Issue by jlambert121** [rodjek/puppet-lint#452]
Putting a parameter with a regexp type in a manifest causes puppet-lint to fail with a "manifests/init.pp:16:syntax:ERROR:Syntax error (try running `puppet parser validate <file>`)" error, but puppet parser validate works fine.

```puppet
Optional[Regexp[/^puppet/]] $source,
```

**Comment by af6140** [comment]
Got the same error.

**Comment by thejandroman** [comment]
Same with Pattern as well:

Example:
```
Pattern[/^\w+?-.*$/] $zone = 'us-east-1',
```
